pkg/crc/machine/machine.go

```diff
@@ -328,13 +330,59 @@ func Ip(ipConfig IpConfig) (IpResult, error) {
 		result.Error = err.Error()
 		return *result, err
 	}
-
 	if result.IP, err = host.Driver.GetIP(); err != nil {
 		result.Success = false
 		result.Error = err.Error()
 		return *result, err
 	}
+	return *result, nil
+}
+
+func Status(statusConfig ClusterStatusConfig) (ClusterStatusResult, error) {
+	result := &ClusterStatusResult{Name: statusConfig.Name, Success: true}
+	api := libmachine.NewClient(constants.MachineBaseDir, constants.MachineCertsDir)
+	_, err := api.Exists(statusConfig.Name)
+	if err != nil {
+		result.Success = false
+		result.Error = err.Error()
+		return *result, err
+	}
+	openshiftStatus := "Stopped"
+	diskUsage := "Unknown"
 
+	host, err := api.Load(statusConfig.Name)
+	if err != nil {
+		result.Success = false
+		result.Error = err.Error()
+		return *result, err
+	}
+	vmStatus, err := host.Driver.GetState()
+	if err != nil {
+		result.Success = false
+		result.Error = err.Error()
+		return *result, err
+	}
+
+	if vmStatus == state.Running {
+		// check if all the clusteroperators are running
+		ocConfig := oc.UseOCWithConfig(statusConfig.Name)
+		cosAvailable, err := oc.GetClusterOperatorStatus(ocConfig)
```

Comment:
imo the "cosAvailable" name is a bit misleading. `oc.GetClusterOperatorStatus(ocConfig)` returns true when all operators are ready. I read `cosAvailable` as something meaning "could we run `oc get co` on the cluster?" (but maybe that's me who is misreading it).

Reply:
Hmm.. would `operatorsRunning` be more meaningful?

Reply:
yes, looks good to me!